modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/snapshot/CurrencySnapshotAcceptanceManager.scala

```diff
@@ -166,6 +178,35 @@ object CurrencySnapshotAcceptanceManager {
           .getOrElse(acc)
       }
 
+    private def acceptFeeTxs(
+      balances: SortedMap[Address, Balance],
+      maybeTxs: Option[SortedSet[Signed[FeeTransaction]]]
+    ): F[(SortedMap[Address, Balance], Option[SortedSet[Signed[FeeTransaction]]])] =
+      maybeTxs match {
+        case None => (balances, maybeTxs).pure[F]
+        case Some(txs) =>
+          txs
+            .foldLeft(balances.view.mapValues(_.value.value).toMap) {
+              case (balances, tx) =>
+                val src = balances.getOrElse(tx.source, Balance.empty.value.value) - tx.amount.value
+                val dst = balances.getOrElse(tx.destination, Balance.empty.value.value) + tx.amount.value
```

Comment:
We need to remember about adding FeeTransaction validation checking that `src` and `dst` are different. In other case it would be possible to generate balances for oneself here. We could also think about an implementation for which it is ambivalent which address is sending to which other address (I guess a list of balance updates - pairs of positive and negative)